src/managers/SessionManager.ts

```diff
@@ -13,36 +13,61 @@ export class SessionManager {
     this.context = context;
   }
 
+  isSafari(): boolean {
+    return typeof window.safari !== "undefined";
+  }
+
+  isHttps(): boolean {
+    return !OneSignalUtils.isUsingSubscriptionWorkaround();
+  }
+
   async notifySWToUpsertSession(
     deviceId: string | undefined,
     deviceRecord: PushDeviceRecord,
     sessionOrigin: SessionOrigin
   ): Promise<void> {
-    Log.debug("Notify SW to upsert session");
+    const isHttps = this.isHttps();
+
     const payload: UpsertSessionPayload = {
       deviceId,
       deviceRecord: deviceRecord.serialize(),
       sessionThreshold: OneSignal.config.sessionThreshold,
       enableSessionDuration: OneSignal.config.enableSessionDuration,
       sessionOrigin,
+      isHttps,
+      isSafari: this.isSafari(),
```

Comment:
why do we need to send isSafari in the session info? Is this a proxy the 'does this platform support service workers/active page events correctly' question?

Reply:
Exactly 'does this browser active page events correctly'. Will rename it, thanks.